### ISSUE TITLE
Scaffold Identity in ASP.NET Core projects

### ISSUE THREAD

**Issue by Byron2016** [dotnet/AspNetCore.Docs#23161]
[**Solution was generated following this steps:**

1.  Create a new project with this template: ASP.NET Core Web App (Model-View-Controller)
2.  Target framework: NET 5.0
3.  Authentication Type: Individuals Accounts
4.  Configure for HTTPS
5.  Enable Razor runtime compilation
6.  With dotnet I installed dotnet-aspnet-codegenerator and Microsoft.VisualStudio.Web.CodeGeneration.Design
7.  I runned command: dotnet aspnet-codegenerator identity -dc MyApplication.Data.ApplicationDbContext

**Section:  Scaffold Identity in ASP.NET Core projects**
**You wrote:** 

> When scaffolding Identity with a new data context into a project with existing individual accounts:
> 
> -  In Startup.ConfigureServices, remove the calls to:
>    - AddDbContext
>    - AddDefaultIdentity
>
>The preceding code comments out the code that is duplicated in Areas/Identity/IdentityHostingStartup.cs

**but that code is not present into Areas/Identity/IdentityHostingStartup.cs**

```
namespace WebApplication1.Areas.Identity
{
    public class IdentityHostingStartup : IHostingStartup
    {
        public void Configure(IWebHostBuilder builder)
        {
            builder.ConfigureServices((context, services) => {
            });
        }
    }
}
```
**Is the tutorial wrong or did I make a mistake?**

]
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 804e24dd-7e30-8cf5-2c99-1ffacb4582f6
* Version Independent ID: 3e25547b-e2c9-6634-2c38-2a77207f17d4
* Content: [Scaffold Identity in ASP.NET Core projects](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/scaffold-identity?view=aspnetcore-5.0&tabs=visual-studio)
* Content Source: [aspnetcore/security/authentication/scaffold-identity.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/authentication/scaffold-identity.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
It's there if you follow the exact directions. It needs to be there to work.
Glad you got it working.

**Comment by Byron2016** [comment]
> It's there if you follow the exact directions. It needs to be there to work.
> Glad you got it working.

Did you follow the 8 steps that I wrote? because I have  file **Areas/Identity/IdentityHostingStartup.cs**  openned and it only have this information:

```
namespace WebApplication1.Areas.Identity
{
    public class IdentityHostingStartup : IHostingStartup
    {
        public void Configure(IWebHostBuilder builder)
        {
            builder.ConfigureServices((context, services) => {
            });
        }
    }
}
```

**Comment by Rick-Anderson** [comment]
So where does it get the DbContext?

**Comment by Byron2016** [comment]
> > When scaffolding Identity with a new data context into a project with existing individual accounts:
> 
> So where does it get the DbContext?

Into startup.cs / ConfigureServices I had to **see below** because file **Areas/Identity/IdentityHostingStartup.cs** does not have the information that you wrote in your manual. 

1. Reemplace 
```
services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
                .AddEntityFrameworkStores<ApplicationDbContext>()
                .AddDefaultTokenProviders();
```
with

```
services.AddIdentity<IdentityUser, IdentityRole>()
                .AddEntityFrameworkStores<ApplicationDbContext>();
```

**Comment by Rick-Anderson** [comment]
fixed in #23968
